anywhere-fitness-fe/src/components/Login.js

```diff
@@ -0,0 +1,161 @@
+import React, {useState, useEffect} from 'react';
+import * as Yup from 'yup';
+import axios from 'axios';
+import styled from 'styled-components'
+
+function Form(props){
+    const defaultState = {
+        email:"",
+        password:"",
+        role:""
+    };
+    const [formState,setFormState]=useState(defaultState);
+
+    const [errors,setErrors]=useState(defaultState)
+
+    const [buttonDisabled,setButtonDisabled]=useState(true);
+
+
+    const formSchema=yup.object().shape({
```

Comment:
![YUP](https://media.giphy.com/media/3oeSAD00YsGzUPTmqA/giphy.gif)

Explain like I'm 5. Whats yup?

Reply:
Instead of creating custom validations for each and every input field, this process can be simplified using Yup. Yup is similar to how we define propTypes in react. We create an object schema using yups object function, we can shape it using the shape method and then we can add in the required method which will need a string which will typically be our error message for a password field for example.